### PR TITLE
Timereports Kind

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: hours-api/src/Futurice/App/HoursApi/Logic.hs [hunk 1]
```diff
@@ -278,14 +279,24 @@ userResponse = User
...
+            TK.KindSickLeave      -> mempty
+            TK.KindBalanceAbsence -> mempty
           where
             NDT hours = fmap realToFrac (report ^. H.timereportAmount) :: NDT 'Hours Float
 
 entryUpdateResponse :: H.MonadHours m => Day -> m EntryUpdateResponse
 entryUpdateResponse d =
     EntryUpdateResponse <$> userResponse <*> hoursResponse (d ... d)
+
+kindToType :: TK.TimereportKind -> EntryType
```
Comment (neurogoo) [inline hours-api/src/Futurice/App/HoursApi/Logic.hs]: I'm a little confused on why to create kind type that is structurally identical to EntryType? Is it just to separete completely Hours from Planmill

Comment (phadej) [reply hours-api/src/Futurice/App/HoursApi/Logic.hs]: it's not structurally identical. `TimereportKind` has more variants.

File: futurice-integrations/src/Futurice/Integrations/TimereportKind.hs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    dutyType <- traverse (`PMQ.enumerationValue` "D") (PM.trDutyType tr)
+
+    pid <- case PM.trProject tr of
+        Just pid -> return (Just pid)
+        Nothing  -> PM.taskProject <$> PMQ.task (PM.trTask tr)
+
+    mproject <- traverse PMQ.project pid
+    maccount <- traverse PMQ.account (mproject >>= PM.pAccount)
+
+    mprojectCategory <- traverse (`PMQ.enumerationValue` "C?") (PM.pCategory <$> mproject)
```
Comment (neurogoo) [inline futurice-integrations/src/Futurice/Integrations/TimereportKind.hs]: What is "C?" or "T?"? If I understand correctly, it is defaultvalue, but feels like a weird value.

Comment (phadej) [reply futurice-integrations/src/Futurice/Integrations/TimereportKind.hs]: just random strings, different from each other. (helps debugging)

Comment (phadej) [reply futurice-integrations/src/Futurice/Integrations/TimereportKind.hs]: C = category, T = type

Comment (phadej) [reply futurice-integrations/src/Futurice/Integrations/TimereportKind.hs]: ... In the longer run we could make `enumerationValue` return `Maybe`, which will be better, but not now.

**Pull request merged**